File: Clase 5/ej1.m
M=1000;

V=normrnd(0,4,1,M);

r=zeros(1,M);

for k=0:(M-1)
    for i=1:(M-k)
        r(1,k+1)=r(1,k+1) + V(1,i)*V(1,i+k);
    end
end
r=r/M;
%r_i=xcorr(V,'biased');
x_1=linspace(1,M,M);
%x_2=linspace(1,length(r_i),length(r_i));
figure
plot(x_1,r);
%plot(x_2,r_i);

x_3=linspace(0,2,M);
PSD=abs(fft(V)).^2;
PSD=PSD/M;
figure
plot(x_3,PSD);




    

File: Clase 6/ej_1.py
import numpy as np
import matplotlib.pyplot as plt
from scipy import signal
from scipy.fft import fft


def auto_corr(y):
    N=len(y)
    r=np.zeros(N)
    for k in range(N-1):
       for i in range(N-k):
            r[k] += y[i]*y[i+k]
    r=r/N
    return r

def PSD_gen(y):
    N=len(y)
    PSD=np.power(np.absolute(y),2)
    PSD=PSD/N
    return PSD

def ej_1():
    #1)
    L=5000

    n=np.linspace(0,L,L)

    x=np.random.normal(0,1,L)
    b=[1,0.5,0.25]
    a=[1]

    x_n=signal.lfilter(b,a,x)

    plt.plot(n,x_n)
    plt.show()
    
 
    r_x_n=auto_corr(x_n)
    plt.plot(n,r_x_n)
    plt.show()
    
    

ej_1()





